src/new.rs

```diff
@@ -95,21 +95,22 @@ pub fn create_new_document(doc_type: &str, name: &str, config: &Config) -> Resul
 
     match doc_type {
         "page" => {
-            println!("page exists? {:?}", Path::new(name).exists());
-            // if !Path::new(name).exists() {
-                try!(create_file(name, index_liquid))
-            // } else {
-                // return Err();
-            // }
+            match Path::new(name).exists() {
+                false => {
+                    info!("Created new {} {}", doc_type, name);
+                    try!(create_file(name, index_liquid))
+                },
+                true => error!("{} already exists", name)
+            };
         },
         "post" => {
-            println!("post exists? {:?}", Path::new(full_path).exists());
-
-        //     if !Path::new(full_path).exists() {
-                try!(create_file(full_path, post_1_md))
-        //     } else {
-        //         error!("here");
-        //     }
+            match Path::new(full_path).exists() {
+                false => {
+                    info!("Created new {} {}", doc_type, name);
+                    try!(create_file(full_path, post_1_md))
+                },
+                true => error!("{} already exists", name)
+            };
         },
```

Comment:
Seems like you could consolidate the error check to be before your `doc_type` `match`.  You could also remove the nested match if you report it as an error rather than "silently" fail (just reporting to logging).

Something like
```rust
if Path::new(name).exists() {
    bail!("{} already exists", full_path);
}

let content = match doc_type {
    "page" => index_liquid,
    "post" => post_1_md,
}

create_file(full_path, content)?;
```

(note: the `page` branch calls `create_file` with `name` and not `full_path`)